stdlib/malloc_dl.c

```diff
@@ -467,18 +467,19 @@ void *malloc(size_t size)
 
 void *calloc(size_t nitems, size_t size)
 {
-	void *ptr;
-	uint64_t allocSize = (uint64_t)nitems * size;
-
-	if (allocSize > (uint64_t)UINT_MAX) {
+	if ((nitems != 0) && (size > (size_t)-1 / nitems)) {
```

Comment:
To avoid signed -> unsigned cast (pedantically it's UB)
```suggestion
	if ((nitems != 0) && (size > (SIZE_MAX / nitems))) {
```